File: api/internal/model/global_waf.go
package model

import "time"

// GlobalWAF is the singleton global WAF default (#198 slice 6). Hosts with
// waf_use_global=true inherit its enabled/mode/paranoia/threshold instead of
// their own proxy_hosts WAF columns. WAF/ModSecurity changes only take effect
// after a proxy container restart (reload does not reparse ModSec rules).
type GlobalWAF struct {
	ID               string    `json:"id"`
	Enabled          bool      `json:"enabled"`
	Mode             string    `json:"mode"` // detection | blocking
	ParanoiaLevel    int       `json:"paranoia_level"`
	AnomalyThreshold int       `json:"anomaly_threshold"`
	CreatedAt        time.Time `json:"created_at"`
	UpdatedAt        time.Time `json:"updated_at"`
}

// UpdateGlobalWAFRequest is the partial-update request for the singleton.
type UpdateGlobalWAFRequest struct {
	Enabled          *bool  `json:"enabled,omitempty"`
	Mode             string `json:"mode,omitempty"`
	ParanoiaLevel    int    `json:"paranoia_level,omitempty"`
	AnomalyThreshold int    `json:"anomaly_threshold,omitempty"`
}

File: ui/src/api/global-waf.ts
import { apiGet, apiPut } from './client'

/**
 * Global WAF default. Mirrors the backend contract at
 * `GET/PUT /api/v1/settings/global-waf`. Hosts with per-host WAF set to
 * "inherit" (waf_use_global=true) follow this default's enabled/mode/paranoia/
 * threshold; updating it regenerates all inheriting hosts server-side. WAF/
 * ModSecurity changes take effect only after a proxy container restart.
 */
export interface GlobalWAF {
  id?: string
  enabled: boolean
  mode: string
  paranoia_level: number
  anomaly_threshold: number
  created_at?: string
  updated_at?: string
}

export interface UpdateGlobalWAFRequest {
  enabled?: boolean
  mode?: string
  paranoia_level?: number
  anomaly_threshold?: number
}

export async function getGlobalWAF(): Promise<GlobalWAF> {
  return apiGet<GlobalWAF>('/api/v1/settings/global-waf')
}

export async function updateGlobalWAF(req: UpdateGlobalWAFRequest): Promise<GlobalWAF> {
  return apiPut<GlobalWAF>('/api/v1/settings/global-waf', req)
}
